File: server/module/module-mall/mall-server/src/convert/mall_promotion_seckill_config.rs
use sea_orm::{Set, NotSet};
use crate::model::mall_promotion_seckill_config::{self, Model as MallPromotionSeckillConfig, ActiveModel as MallPromotionSeckillConfigActiveModel};
use mall_model::request::mall_promotion_seckill_config::{CreateMallPromotionSeckillConfigRequest, UpdateMallPromotionSeckillConfigRequest};
use mall_model::response::mall_promotion_seckill_config::MallPromotionSeckillConfigResponse;

pub fn create_request_to_model(request: &CreateMallPromotionSeckillConfigRequest) -> MallPromotionSeckillConfigActiveModel {
    MallPromotionSeckillConfigActiveModel {
        name: Set(request.name.clone()),
        start_time: Set(request.start_time.clone()),
        end_time: Set(request.end_time.clone()),
        slider_pic_urls: Set(request.slider_pic_urls.clone()),
        status: Set(request.status.clone()),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateMallPromotionSeckillConfigRequest, existing: MallPromotionSeckillConfig) -> MallPromotionSeckillConfigActiveModel {
    let mut active_model: MallPromotionSeckillConfigActiveModel = existing.into();
    if let Some(name) = &request.name { 
        active_model.name = Set(name.clone());
    }
    if let Some(start_time) = &request.start_time { 
        active_model.start_time = Set(start_time.clone());
    }
    if let Some(end_time) = &request.end_time { 
        active_model.end_time = Set(end_time.clone());
    }
    if let Some(slider_pic_urls) = &request.slider_pic_urls { 
        active_model.slider_pic_urls = Set(slider_pic_urls.clone());
    }
    if let Some(status) = &request.status { 
        active_model.status = Set(status.clone());
    }
    active_model
}

pub fn model_to_response(model: MallPromotionSeckillConfig) -> MallPromotionSeckillConfigResponse {
    MallPromotionSeckillConfigResponse { 
        id: model.id,
        name: model.name,
        start_time: model.start_time,
        end_time: model.end_time,
        slider_pic_urls: model.slider_pic_urls,
        status: model.status,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

File: admin/src/api/mall_promotion_seckill_config.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/mall/mall_promotion_seckill_config/create', // 新增
  update: '/mall/mall_promotion_seckill_config/update', // 修改
  delete: '/mall/mall_promotion_seckill_config/delete', // 删除
  get: '/mall/mall_promotion_seckill_config/get', // 单条查询
  list: '/mall/mall_promotion_seckill_config/list', // 列表查询
  page: '/mall/mall_promotion_seckill_config/page', // 分页查询
  enable: '/mall/mall_promotion_seckill_config/enable', // 启用
  disable: '/mall/mall_promotion_seckill_config/disable', // 禁用
}

export interface MallPromotionSeckillConfigRequest {
  id: number; // 编号
  name: string; // 秒杀时段名称
  start_time: string; // 开始时间点
  end_time: string; // 结束时间点
  slider_pic_urls: string; // 秒杀主图
  status: number; // 活动状态
  }

export interface MallPromotionSeckillConfigResponse {
  id: number; // 编号
  name: string; // 秒杀时段名称
  start_time: string; // 开始时间点
  end_time: string; // 结束时间点
  slider_pic_urls: string; // 秒杀主图
  status: number; // 活动状态
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
  }

export interface MallPromotionSeckillConfigQueryCondition extends PaginatedRequest {
  
}

export const createMallPromotionSeckillConfig = (mall_promotion_seckill_config: MallPromotionSeckillConfigRequest): Promise<number> => {
  return api.post<number>(apis.create, mall_promotion_seckill_config);
}

export const updateMallPromotionSeckillConfig = (mall_promotion_seckill_config: MallPromotionSeckillConfigRequest): Promise<void> => {
  return api.post<void>(apis.update, mall_promotion_seckill_config);
}

export const deleteMallPromotionSeckillConfig = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getMallPromotionSeckillConfig = (id: number): Promise<MallPromotionSeckillConfigResponse> => {
  return api.get<MallPromotionSeckillConfigResponse>(`${apis.get}/${id}`);
}

export const listMallPromotionSeckillConfig = (): Promise<Array<MallPromotionSeckillConfigResponse>> => {
  return api.get<Array<MallPromotionSeckillConfigResponse>>(apis.list);
}

export const pageMallPromotionSeckillConfig = (condition: MallPromotionSeckillConfigQueryCondition): Promise<PaginatedResponse<MallPromotionSeckillConfigResponse>> => {
  return api.get<PaginatedResponse<MallPromotionSeckillConfigResponse>>(apis.page, condition);
}

export const enableMallPromotionSeckillConfig = (id: number): Promise<void> => {
  return api.post<void>(`${apis.enable}/${id}`);
}

export const disableMallPromotionSeckillConfig = (id: number): Promise<void> => {
  return api.post<void>(`${apis.disable}/${id}`);
}